src/daft-logical-plan/src/optimization/rules/simplify_expressions.rs

```diff
@@ -0,0 +1,512 @@
+        // A / 1 --> A
+        Expr::BinaryOp {
+            op: Operator::TrueDivide,
+            left,
+            right,
+        } if is_one(&right) => Transformed::yes(left),
+        // null / A --> null
+        Expr::BinaryOp {
+            op: Operator::TrueDivide,
+            left,
+            right: _,
+        } if is_null(&left) => Transformed::yes(left),
+        // A / null --> null
+        Expr::BinaryOp {
+            op: Operator::TrueDivide,
+            left: _,
+            right,
+        } if is_null(&right) => Transformed::yes(right),
+
+        // ----------------
+        // Addition
+        // ----------------
+        // A + 0 --> A
+        Expr::BinaryOp {
+            op: Operator::Plus,
+            left,
+            right,
+        } if is_zero(&right) => Transformed::yes(left),
+
+        // 0 + A --> A
+        Expr::BinaryOp {
+            op: Operator::Plus,
+            left,
+            right,
+        } if is_zero(&left) => Transformed::yes(right),
+
+        // ----------------
+        // Subtraction
+        // ----------------
+
+        // A - 0 --> A
+        Expr::BinaryOp {
+            op: Operator::Minus,
+            left,
+            right,
+        } if is_zero(&right) => Transformed::yes(left),
+
+        // A - null --> null
+        Expr::BinaryOp {
+            op: Operator::Minus,
+            left: _,
+            right,
+        } if is_null(&right) => Transformed::yes(right),
+        // null - A --> null
+        Expr::BinaryOp {
+            op: Operator::Minus,
+            left,
+            right: _,
+        } if is_null(&left) => Transformed::yes(left),
+
+        // ----------------
+        // Modulus
+        // ----------------
+
+        // A % null --> null
+        Expr::BinaryOp {
+            op: Operator::Modulus,
+            left: _,
+            right,
+        } if is_null(&right) => Transformed::yes(right),
+
+        // null % A --> null
+        Expr::BinaryOp {
+            op: Operator::Modulus,
+            left,
+            right: _,
+        } if is_null(&left) => Transformed::yes(left),
+
+        // A BETWEEN low AND high --> A >= low AND A <= high
+        Expr::Between(expr, low, high) => {
+            let expr = simplify_expr(Arc::unwrap_or_clone(expr), schema)?.data;
+            let low = simplify_expr(Arc::unwrap_or_clone(low), schema)?.data;
+            let high = simplify_expr(Arc::unwrap_or_clone(high), schema)?.data;
+            Transformed::yes(expr.clone().lt_eq(high).and(expr.gt_eq(low)))
+        }
+        Expr::Not(expr) => match Arc::unwrap_or_clone(expr) {
+            // NOT (BETWEEN A AND B) --> A < low OR A > high
+            Expr::Between(expr, low, high) => {
+                let expr = simplify_expr(Arc::unwrap_or_clone(expr), schema)?.data;
+                let low = simplify_expr(Arc::unwrap_or_clone(low), schema)?.data;
+                let high = simplify_expr(Arc::unwrap_or_clone(high), schema)?.data;
```

Comment:
Instead of doing the recursion yourself, you should use `TreeNode::transform_down` or `TreeNode::transform_up` (I would probably say the latter so you can propagate constants up a tree). That way, children of expressions that are not `Expr::Between` or `Expr::Not` can also be simplified